main.go

```diff
@@ -563,3 +568,56 @@ func deployCloudComponents(plan types.Plan) error {
 
 	return nil
 }
+
+func featureEnabled(features []string, secretFeatures []string) bool {
+	for _, feature := range features {
+		for _, secretFeature := range secretFeatures {
+			if feature == secretFeature {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func filterFeatures(plan types.Plan) types.Plan {
+	plan.Features = append(plan.Features, types.DefaultFeature)
+
+	if (plan.S3 != types.S3{}) {
+		plan.Features = append(plan.Features, types.S3Bucket)
+	}
+
+	if (plan.Github != types.Github{}) {
+		plan.Features = append(plan.Features, types.GitHub)
+	}
+
+	if plan.TLS == true {
+		plan = filterDNSFeature(plan)
+	}
+
+	if plan.EnableOAuth == true {
+		plan.Features = append(plan.Features, types.Auth)
+	}
+
+	if plan.BasicAuth == true {
```

Comment:
This should be mandatory. 

There are things which are optional/configurable and things which are not.

Not:
s3
payload_secret
registry and so on

Optional
tls
auth

Configurable
type of auth / tls / scm (at least one scm)

Reply:
Removed `registry` and `basic_auth` as those are mandatory 👍